dot/services.go

```diff
@@ -321,12 +333,24 @@ func (nodeBuilder) createRPCService(params rpcServiceSettings) (*rpc.HTTPServer,
 	)
 	rpcService := rpc.NewService()
 
+	runtime, err := params.state.Block.GetRuntime(nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting runtime: %w", err)
+	}
+
+	stateVersion, err := trie.ParseVersion(runtime.Version().StateVersion)
+	if err != nil {
+		return nil, fmt.Errorf("parsing state version: %w", err)
+	}
+
 	genesisData, err := params.state.Base.LoadGenesisData()
 	if err != nil {
 		return nil, fmt.Errorf("failed to load genesis data: %s", err)
 	}
 
-	syncStateSrvc, err := modules.NewStateSync(genesisData, params.state.Storage)
+	// TODO should we inject the state version from the latest runtime?
+	// Or somehow find the genesis state version???
+	syncStateSrvc, err := modules.NewStateSync(genesisData, params.state.Storage, stateVersion)
```

Comment:
TODO

Reply:
From Kishan

> I feel like it would be best block runtime
> substrate has features to update the runtime. So, with that in mind taking info from best block might make more sense.

From Eclesio

> Given the current substrate implementation you should use the latest finalized block -> https://github.com/paritytech/substrate/blob/eddf888394cba28b048bd79e319e8b27f4b414f3/client/sync-state-rpc/src/lib.rs#L142
> So you can use the finalized block runtime